x/ibc/20-transfer/types/msgs.go

```diff
@@ -73,3 +74,59 @@ func (msg MsgTransfer) GetSignBytes() []byte {
 func (msg MsgTransfer) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Sender}
 }
+
+type MsgTransfer struct {
+	SourcePort    string         `json:"source_port" yaml:"source_port"`       // the port on which the packet will be sent
+	SourceChannel string         `json:"source_channel" yaml:"source_channel"` // the channel by which the packet will be sent
+	Amount        sdk.Coins      `json:"amount" yaml:"amount"`                 // the tokens to be transferred
+	Sender        sdk.AccAddress `json:"sender" yaml:"sender"`                 // the sender address
+	Receiver      sdk.AccAddress `json:"receiver" yaml:"receiver"`             // the recipient address on the destination chain
+	Source        bool           `json:"source" yaml:"source"`                 // indicates if the sending chain is the source chain of the tokens to be transferred
+}
+
+// NewMsgRecvPacket creates a new MsgRecvPacket instance
+func NewMsgRecvPacket(packet channel.PacketI, proofs []commitment.Proof, height uint64, signer sdk.AccAddress) MsgRecvPacket {
+	return MsgRecvPacket{
+		Packet: packet,
+		Proofs: proofs,
+		Height: height,
+		Signer: signer,
+	}
+}
+
+// Route implements sdk.Msg
+func (MsgRecvPacket) Route() string {
+	return RouterKey
+}
+
+// Type implements sdk.Msg
+func (MsgRecvPacket) Type() string {
+	return "recv_packet"
+}
+
+// ValidateBasic implements sdk.Msg
+func (msg MsgRecvPacket) ValidateBasic() sdk.Error {
+	if msg.Proofs == nil {
```

Comment:
we need to also iterate over the proofs and check if any of them is `nil`

Reply:
Added.